framework/CodeInterfaces/MELCOR/MELCORdata.py

```diff
@@ -13,62 +13,54 @@
 # limitations under the License.
 """
   Created on April 18, 2017
-  @author: Matteo D'Onorio (University of Rome La Sapienza),
+  @author: Matteo Donorio (University of Rome La Sapienza),
            Fabio Gianneti (University of Rome La Sapienza),
            Andrea Alfonsi (INL)
-
-  Modified on January 24, 2018
-  @author: Violet Olson
-           Thomas Riley (Oregon State University)
-           Robert Shannon (Oregon State University)
-  Change Summary: Added Control Function parsing
-
-  Modified on July 4, 2018
-  @author: Matteo D'Onorio (University of Rome La Sapienza)
-  Change Summary: Added Function to write CSV file from PTF
-                  All the functions for the parsing have been removed
-  
 """
 from __future__ import division, print_function, unicode_literals, absolute_import
 from melcor_tools import MCR_bin
+from utils import utils
 import pandas as pd
 import warnings
 warnings.simplefilter('default',DeprecationWarning)
+import os
 import re
 import copy
+import fileinput
+from BaseClasses import BaseType
 
-class MELCORdata:
+class MELCORdata():
   """
-    class that parses output of MELCOR 2.1 output file and reads in trip, minor block and write a csv file
-    For now, Only the data associated to control volumes and control functions are parsed and output
+    This class is the CodeInterface for MELGEN (a sub-module of Melcor)
   """
-  def __init__(self,filen):
+  
+  def __init__(self,origInputFiles):
     """
       Constructor
-      @ In, filen, FileObject, the file to parse
+      @ In, None
       @ Out, None
     """
-    self.lines      = open(filen,"r").readlines()
-    timeBlocks      = self.getTimeBlocks()
-    self.timeParams = {}
-    volForEachTime  = self.returnVolumeHybro(timeBlocks)
-    self.timeParams.update(volForEachTime)
-    self.functions  = self.returnControlFunctions(timeBlocks)
-
-  def writeCsv(self,filen,filen2):
+    self.printTag = "MELCOR PARSER"
+    os.chdir(self.runInfoDict['WorkingDir'])
+    #self.inputFiles = inputFiles
+    workingDir = os.path.join(os.path.dirname(__file__), '..')
+    Input = os.path.join(workingDir,'prova.i')
+    f = open(Input)
+    self.VarSrch = MelcorApp.VarList
+    self.MELCORPlotFile = MelcorApp.MelcorPlotFile
+    
+    
+  def writeCsv(self,filen,workDir):   
     """
       Output the parsed results into a CSV file
       @ In, filen, str, the file name of the CSV file
+      @ In, workDir, str, current working directory
       @ Out, None
-    """
+    """        
```

Comment:
remove whitespaces 

runt the  script "./developer_tools/delete_trailing_whitespace.sh framework"  in the raven folder